colcon_ros/task/cmake/build.py

```diff
@@ -28,4 +30,18 @@ def __init__(self):  # noqa: D107
         extension = CmakeBuildTask_()
         extension.set_context(context=self.context)
 
+        # Add extra CMake variables to help packages detect ROS
+        if self.context.args.cmake_args is None:
+            self.context.args.cmake_args = []
+        self.context.args.cmake_args += ['-D' + 'COLCON_VERSION:STRING='
```

Comment:
This one isn't ROS specific, and might make more sense to put in  in [https://github.com/colcon/colcon-cmake](`colcon-cmake`)

https://github.com/colcon/colcon-cmake/blob/7d5bbeafc2e52b979c93e5c0215d805744eaf3c1/colcon_cmake/task/cmake/build.py#L144

Reply:
Also, it might make sense for the type to be `INTERNAL` instead of `STRING` since this isn't expected to be changed by a user

Reply:
The idea is to have ROS specific variables that can be detected by the CMake build. Setting them to INTERNAL is fine.